staging/src/k8s.io/legacy-cloud-providers/gce/gce_loadbalancer_internal.go

```diff
@@ -512,6 +514,17 @@ func (g *Cloud) ensureInternalInstanceGroup(name, zone string, nodes []*v1.Node)
 		kubeNodes.Insert(n.Name)
 	}
 
+	// Individual InstanceGroup has a limit for 1000 instances in it.
+	// As a result, it's not possible to add more to it.
+	// Given that the long-term fix (AlphaFeatureILBSubsets) is already in-progress,
+	// to stop the bleeding we now simply cut down the contents to first 1000
+	// instances in the alphabetical order. Since there is a limitation for
+	// 250 backend VMs for ILB, this isn't making things worse.
+	if len(kubeNodes) > maxInstancesPerInstanceGroup {
+		klog.Warningf("Limiting number of VMs for InstanceGroup %s to %d", name, maxInstancesPerInstanceGroup)
+		kubeNodes = sets.NewString(kubeNodes.List()[:maxInstancesPerInstanceGroup]...)
```

Comment:
The node list is passed in from https://github.com/kubernetes/kubernetes/blob/64d75b3bad5b153df0e95c9f8fd8337090096a49/pkg/controller/service/controller.go#L389

Is this truncation assumed to be in alphabetical order because the nodelister returns the nodes in alphabetical order? If not, this might pick a different list of 1000 nodes in ever update. Can we add a comment here?

Reply:
sets.List() is returning items sorted:
https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apimachinery/pkg/util/sets/string.go#L171

Reply:
ok, looks good.